lib/package.js

```diff
@@ -471,9 +471,21 @@ exports.download = function(pkg, options, installDeps) {
 
         // get the override
         if (endpoint.getOverride) {
-          if (!override && config.pjson.overrides[pkg.exactName])
-            ui.log('warn', 'Using local override for `' + pkg.exactName + '`');
-          return endpoint.getOverride(pkg.registry, pkg.package, pkg.version, override || config.pjson.overrides[pkg.exactName]);
+          if (!override) {
+            for (var overrideName in config.pjson.overrides) {
+              if (overrideName.indexOf(pkg.name) !== -1) {
+                var overrideVersion = overrideName.split('@')[1];
+
+                // use an override if it is compatible per semver
+                if (semver.match(overrideVersion, pkg.version)) {
```

Comment:
This should be `semver.match('^' + overrideVersion)`. The idea is to make the override range semver-based just like the registry, not to allow arbitrary version ranges for overrides. Alternatively do you think we should just do the version ranges and update the registry to be based on version ranges for overrides as well?